packages/plugins/rpc-manager/src/lib/eth_sendRawTransaction.ts

```diff
@@ -0,0 +1,138 @@
+// TODO: 3. add only 1 instance of registerActionForEvent('blockchain:proxy:request/response')
+// TODO: 4. For each request/response, loop through registered RPC modifiers finding matches against RPC method name
+// TODO: 5. run matched action for request/response
+// TODO: This should be done after https://github.com/embarklabs/embark/pull/2150 is merged.
+export default class EthSendRawTransaction extends RpcModifier {
+  constructor(embark: Embark, rpcModifierEvents: EmbarkEvents) {
+    super(embark, rpcModifierEvents);
+
+    embark.registerActionForEvent("blockchain:proxy:request", this.ethSendRawTransactionRequest.bind(this));
+    embark.registerActionForEvent("blockchain:proxy:response", this.ethSendRawTransactionResponse.bind(this));
+  }
+
+  private stripPrefix(value) {
+    return value?.indexOf('0x') === 0 ? value.substring(2) : value;
+  }
+
+  private async ethSendRawTransactionRequest(params: any, callback: Callback<any>) {
+    if (params.request.method !== blockchainConstants.transactionMethods.eth_sendRawTransaction) {
+      return callback(null, params);
+    }
+
+    const accounts = await this.accounts;
+    if (!(accounts && accounts.length)) {
+      return callback(null, params);
+    }
+
+    // Only handle quorum requests that came via eth_sendTransaction
+    // If the user wants to send a private raw tx directly,
+    // web3.eth.sendRawPrivateTransaction should be used (which calls
+    // eth_sendRawPrivateTransaction)
+    const originalPayload = params.originalRequest?.params[0];
+    const privateFor = originalPayload?.privateFor;
+    const privateFrom = originalPayload?.privateFrom;
+    if (!privateFor && !privateFrom) {
+      return callback(null, params);
+    }
+
+    const account = accounts.find((acc) => Web3.utils.toChecksumAddress(acc.address) === Web3.utils.toChecksumAddress(originalPayload.from));
+    if (!account?.privateKey) {
+      return callback(null, params);
+    }
+
+    // manually send to the node in the response
+    params.sendToNode = false;
+    callback(null, params);
+  }
+
+  private async ethSendRawTransactionResponse(params: any, callback: Callback<any>) {
+    if (params.request.method !== blockchainConstants.transactionMethods.eth_sendRawTransaction) {
+      return callback(null, params);
+    }
+
+    const accounts = await this.accounts;
+    if (!(accounts && accounts.length)) {
+      return callback(null, params);
+    }
+
+    // Only handle quorum requests that came via eth_sendTransaction
+    // If the user wants to send a private raw tx directly,
+    // web3.eth.sendRawPrivateTransaction should be used (which calls
+    // eth_sendRawPrivateTransaction)
+    const originalPayload = params.originalRequest?.params[0];
+    const privateFor = originalPayload?.privateFor;
+    const privateFrom = originalPayload?.privateFrom;
+    if (!privateFor && !privateFrom) {
+      return callback(null, params);
+    }
```

Comment:
The start of the function is the same for both functions. Could be nice to extract it to a small utility function just above.

Reply:
Good idea, this simplified the functions quite a lot.